packages/dot-agent-kit/src/dot_agent_kit/data/kits/erk/scripts/erk/exit_plan_mode_hook.py

```
@@ -136,31 +136,63 @@ def _find_session_plan(session_id: str) -> Path | None:
     return None
 
 
+def _get_current_branch() -> str | None:
+    """Get the current git branch name.
+
+    Returns:
+        Branch name, or None if not in a git repo or detached HEAD
+    """
+    try:
+        result = subprocess.run(
+            ["git", "rev-parse", "--abbrev-ref", "HEAD"],
+            capture_output=True,
+            text=True,
+            check=True,
+        )
+        return result.stdout.strip()
+    except subprocess.CalledProcessError:
+        return None
+
```

allow this to fail hard. do not catch